runtime/src/pdk.rs

```diff
@@ -132,6 +122,22 @@ pub(crate) fn var_set(
         None => anyhow::bail!("invalid handle offset for var value: {voffset}"),
     };
 
+    let mut size = std::mem::size_of::<String>()
+        + std::mem::size_of::<Vec<u8>>()
+        + key.len()
+        + handle.length as usize;
+
+    for (k, v) in data.vars.iter() {
+        size += k.len();
+        size += v.len();
+        size += std::mem::size_of::<String>() + std::mem::size_of::<Vec<u8>>();
+    }
+
+    // If the store is larger than the configured size, or 1mb by default, then stop adding things
+    if size > data.manifest.memory.max_var_bytes.unwrap_or(1024 * 1024) as usize && voffset != 0 {
```

Comment:
would it make sense to add a check for `data.manifest.memory.max_var_bytes.unwrap_or(1024 * 1024) as usize == 0` at the top of this function, with an error of something like `Vars are disabled by this host`, just to short-circuit this and leave a more clear error? 

i _think_ that the only error someone will see here is `Variable store is full`, which doesn't tell me if Vars are strictly disabled, which would mean I need to change my code as a plugin author. on the other hand, it also leaks some host-specific information to the plugin, which is probably ok?

Reply:
good idea, i will add that so the user can get a more helpful error message!